src/arbitrary.rs

```diff
@@ -0,0 +1,129 @@
+                if u.arbitrary::<bool>()? {
+                    let mut bytes = vec![0u8; P2WPKH_LEN];
+                    u.fill_buffer(&mut bytes)?;
+                    bytes
+                } else {
+                    let mut bytes = vec![0u8; P2WSH_LEN];
+                    u.fill_buffer(&mut bytes)?;
+                    bytes
+                }
+            }
+            4 => {
+                // P2TR: Generate a valid X-only public key
+                let mut secret_bytes = [0u8; 32];
+                u.fill_buffer(&mut secret_bytes)?;
+                // Keep trying until we get a valid key
+                while SecretKey::from_slice(&secret_bytes).is_err() {
+                    u.fill_buffer(&mut secret_bytes)?;
+                }
+                let secret_key = SecretKey::from_slice(&secret_bytes).unwrap();
+                let keypair = Keypair::from_secret_key(SECP256K1, &secret_key);
+                let (x_only_pub_key, _parity) = keypair.x_only_public_key();
+                let bytes = x_only_pub_key.serialize().to_vec();
+                assert_eq!(bytes.len(), P2TR_LEN);
+                bytes
+            }
+            _ => unreachable!(),
+        };
+
+        // Construct the descriptor
+        let mut bytes = Vec::with_capacity(1 + payload.len());
+        bytes.push(type_tag);
+        bytes.extend_from_slice(&payload);
+
+        Descriptor::from_bytes(&bytes).map_err(|_| Error::IncorrectFormat)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use arbitrary::Arbitrary;
+    use bitcoin::Network;
+    use rand_core::{OsRng, RngCore};
+
+    use crate::DescriptorType;
+
+    #[test]
+    fn test_arbitrary_descriptor() {
+        // Create a buffer of random bytes
+        let mut data = vec![0u8; 1024];
+        OsRng.fill_bytes(&mut data);
+        let mut u = Unstructured::new(&data);
+
+        // Generate several random descriptors
+        for _ in 0..1_000 {
+            if let Ok(desc) = Descriptor::arbitrary(&mut u) {
+                // Verify type tag is valid
+                match desc.type_tag() {
+                    DescriptorType::OpReturn => assert!(desc.payload().len() <= MAX_OP_RETURN_LEN),
+                    DescriptorType::P2pkh => assert_eq!(desc.payload().len(), P2PKH_LEN),
+                    DescriptorType::P2sh => assert_eq!(desc.payload().len(), P2SH_LEN),
+                    DescriptorType::P2wpkh => assert_eq!(desc.payload().len(), P2WPKH_LEN),
+                    DescriptorType::P2wsh => assert_eq!(desc.payload().len(), P2WSH_LEN),
+                    DescriptorType::P2tr => assert_eq!(desc.payload().len(), P2TR_LEN),
+                }
+            }
+        }
+    }
+
+    #[test]
+    fn test_arbitrary_descriptor_to_address() {
+        // Create a buffer of random bytes
+        let mut data = vec![0u8; 1024];
+        OsRng.fill_bytes(&mut data);
+        let mut u = Unstructured::new(&data);
```

Comment:
This isn't enough data for the subsequent loop.

Reply:
Done in 252a8112bf384202777d5149d6bcb18767a9116f